npm/pkg/dataplane/dataplane_windows.go

```diff
@@ -1,33 +1,164 @@
 package dataplane
 
 import (
-	"github.com/Azure/azure-container-networking/npm"
+	"fmt"
+
+	"github.com/Azure/azure-container-networking/npm/pkg/dataplane/policies"
 	"github.com/Microsoft/hcsshim/hcn"
 	"k8s.io/klog"
 )
 
-const (
-	// Windows specific constants
-	AzureNetworkName = "azure"
-)
-
 // initializeDataPlane will help gather network and endpoint details
 func (dp *DataPlane) initializeDataPlane() error {
-	klog.Infof("Initializing dataplane for windows")
+	klog.Infof("[DataPlane] Initializing dataplane for windows")
 
-	// Get Network ID
-	network, err := hcn.GetNetworkByName(AzureNetworkName)
+	err := dp.setNetworkIDByName(AzureNetworkName)
 	if err != nil {
 		return err
 	}
 
-	dp.networkID = network.Id
+	err = dp.refreshAllPodEndpoints()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (dp *DataPlane) shouldUpdatePod() bool {
+	return true
+}
+
+// updatePod has two responsibilities in windows
+// 1. Will call into dataplane and updates endpoint references of this pod.
+// 2. Will check for existing applicable network policies and applies it on endpoint
+func (dp *DataPlane) updatePod(pod *UpdateNPMPod) error {
```

Comment:
what is this struct?

Reply:
When a pod add/update event happens, a PodIP gets added to multiple IPsets and can be removed from multiple Ips, this struct helps provide all those changes in one go so dataplane can determine if any network policies should be removed or applied on this pods endpoints.